app/models/piece.rb

```diff
@@ -1,9 +1,62 @@
 class Piece < ApplicationRecord
   belongs_to :game
+
+  PAWN = 'Pawn'.freeze
+  ROOK = 'Rook'.freeze
+  KNIGHT = 'Knight'.freeze
+  BISHOP = 'Bishop'.freeze
+  QUEEN = 'Queen'.freeze
+  KING = 'King'.freeze
+
+  def obstructed?(destination_x, destination_y)
+    return true if self.invalid?(destination_x, destination_y)
+    return true if self.horizontal_or_vertical_obstruction?(destination_x, destination_y)
+    return true if self.diagonal_obstruction?(destination_x, destination_y)
+    false
+  end
+
+  def horizontal_or_vertical?(destination_x, destination_y)
+    return false unless destination_x == self.x_position || destination_y == self.y_position
```

Comment:
In Ruby. Methods will always return the value of whatever is executed last in a method.

Because of this, you don't need to use `return` if it's the last line.

Also, since `destination_x == self.x_position || destination_y == self.y_position` evaluates to `true` or `false`. You can remove the `false unless`.